src/shared/sleep-config.c

```diff
@@ -271,6 +327,37 @@ int put_battery_discharge_rate(int estimated_battery_discharge_rate) {
         return 0;
 }
 
+/* Estimate battery discharge rate using stored previous and current capacity over timestamp difference */
+int estimate_battery_discharge_rate_per_hour(
+                                        Hashmap *last_capacity,
+                                        Hashmap *current_capacity,
+                                        usec_t before_timestamp,
+                                        usec_t after_timestamp) {
+        _cleanup_(sd_device_enumerator_unrefp) sd_device_enumerator *e = NULL;
+        uint64_t system_hash_id;
+        sd_device *dev;
+        int *battery_last_capacity, *battery_current_capacity, battery_discharge_rate, r;
+
+        get_sd_device_battery(e);
+        FOREACH_DEVICE(e, dev) { /* todo: return -ENOENT for no battery dev and skip AC check*/
+                battery_last_capacity = hashmap_get(last_capacity, dev);
+                battery_current_capacity = hashmap_get(current_capacity, dev);
```

Comment:
These should use `PTR_TO_INT(hashmap_get(...))` to get the value out of the hashmap.

Reply:
ahh yes right. i was wondering why the value is not fetched. thanks. fixed.